src/modules/RfqModule.sol

```diff
@@ -0,0 +1,196 @@
+// SPDX-License-Identifier: GPL-3.0-only
+pragma solidity ^0.8.13;
+
+// Libraries
+import "openzeppelin/utils/math/SafeCast.sol";
+import "lyra-utils/decimals/SignedDecimalMath.sol";
+import "lyra-utils/decimals/DecimalMath.sol";
+
+// Inherited
+import {BaseModule} from "./BaseModule.sol";
+import {IRfqModule} from "../interfaces/IRfqModule.sol";
+
+// Interfaces
+import {IBaseManager} from "v2-core/src/interfaces/IBaseManager.sol";
+import {IDataReceiver} from "v2-core/src/interfaces/IDataReceiver.sol";
+import {ISubAccounts} from "v2-core/src/interfaces/ISubAccounts.sol";
+import {IAsset} from "v2-core/src/interfaces/IAsset.sol";
+import {IPerpAsset} from "v2-core/src/interfaces/IPerpAsset.sol";
+import {IMatching} from "../interfaces/IMatching.sol";
+
+import "forge-std/console2.sol";
+
+/**
+ * @title RfqModule
+ * @dev Allows a user to request a bundle of trades to all be executed atomically
+ */
+contract RfqModule is IRfqModule, BaseModule {
+  using SafeCast for uint;
+  using SafeCast for int;
+  using SignedDecimalMath for int;
+  using DecimalMath for uint;
+
+  // @dev we fix the quoteAsset for the contracts so we only support one quoteAsset per deployment
+  IAsset public immutable quoteAsset;
+
+  mapping(IPerpAsset => bool) public isPerpAsset;
+
+  uint public feeRecipient;
+
+  constructor(IMatching _matching, IAsset _quoteAsset, uint _feeRecipient) BaseModule(_matching) {
+    quoteAsset = _quoteAsset;
+    feeRecipient = _feeRecipient;
+  }
+
+  ////////////////////
+  //   Owner-Only   //
+  ////////////////////
+
+  /**
+   * @dev set fee recipient account
+   */
+  function setFeeRecipient(uint _feeRecipient) external onlyOwner {
+    feeRecipient = _feeRecipient;
+  }
+
+  /**
+   * @dev set perp asset mapping
+   */
+  function setPerpAsset(IPerpAsset _perpAsset, bool isPerp) external onlyOwner {
+    isPerpAsset[_perpAsset] = isPerp;
+  }
+
+  ////////////////////////
+  //   Action Handler   //
+  ////////////////////////
+
+  /**
+   * @dev Assumes VerifiedActions are sorted in the order: [requesterAction, takerActions]
+   * @param actions The actions to execute
+   * @param actionDataBytes The data to pass to the module by the executor. Expected to be OrderData
+   */
+  function executeAction(VerifiedAction[] memory actions, bytes memory actionDataBytes)
+    external
+    onlyMatching
+    returns (uint[] memory newAccIds, address[] memory newAccOwners)
+  {
+    // Verify
+    if (actions.length != 2) revert RFQM_InvalidActionsLength();
+
+    // Maker (user placing RFQ order) is always receiving the assets from the taker (user filling order)
```

Comment:
"always receiving the assets from the taker" - is this true? thought your comment above states makers are not necessarily buying